### PR TITLE
Bugfix/arsn 329 bump mongodb driver

### PR DESCRIPTION
**Pull request by ghost**
Bump mongodb related dependencies (driver and memory server) to latest version, and making the changes necessary. No breaking change introduced, it only dapts the mongodb usage to switch from callback-based to promise based, as highly recommended by mongodb documentation.

See: https://www.mongodb.com/docs/drivers/node/current/compatibility/

This change is compatible with our current and future mongodb versions.

Changes are mostly the following ones:
- Switch from callback (removed in v5) to promises
- Use atomic operations in bulk
- Remove deprecated mongodb functions like `update` (becomes `updateOne`)
- Adapt the tests with the changes, as some are dependent on the mongodb-memory-server version (or mongodb driver version)

### CODE REVIEW

File: tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js
Comment (KillianG) [inline tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js:7]: nit: why don't you put that line 6 ? because of linter I guess ? asking cause line 10 is longer and in a single line

**Review by KillianG — APPROVED** [review]

File: tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js
Comment (williamlardier) [reply tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js:7]: Yes, the linter (or my IDE?) changed it, it could go into the same line though, as you said. I'll try to do it before merging (I'll wait for other reviews first).

Comment (ghost) [reply tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js:7]: Yes, the linter (or my IDE?) changed it, it could go into the same line though, as you said. I'll try to do it before merging (I'll wait for other reviews first).

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**
